homeassistant/components/qbus/config_flow.py

```diff
@@ -0,0 +1,159 @@
+"""Config flow for Qbus."""
+
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, Any
+
+from qbusmqttapi.discovery import QbusMqttDevice
+from qbusmqttapi.factory import QbusMqttMessageFactory, QbusMqttTopicFactory
+
+from homeassistant import config_entries
+from homeassistant.components.mqtt import client as mqtt
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.helpers.service_info.mqtt import MqttServiceInfo
+
+from .const import CONF_ID, CONF_SERIAL, DOMAIN
+from .qbus import QbusConfigContainer
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class QbusFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle Qbus config flow."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize."""
+        self._message_factory = QbusMqttMessageFactory()
+        self._topic_factory = QbusMqttTopicFactory()
+
+        self._gateway_topic = self._topic_factory.get_gateway_state_topic()
+        self._config_topic = self._topic_factory.get_config_topic()
+        self._controller_topic = self._topic_factory.get_device_state_topic("+")
```

Comment:
I see "device" and "controller" terms being used for the same thing. Is a device a controller? Should we use only "device" and drop the "controller" term? or they are different things?

Reply:
Now "device" is being used consistently, except for UI stuff. Users know it better as a "controller".